dynamic-security-analysis/src/main/java/com/powsybl/security/dynamic/DynamicSecurityAnalysis.java

```diff
@@ -0,0 +1,295 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.security.dynamic;
+
+import com.powsybl.action.Action;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.Versionable;
+import com.powsybl.commons.config.PlatformConfig;
+import com.powsybl.commons.config.PlatformConfigNamedProvider;
+import com.powsybl.commons.report.ReportNode;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.computation.local.LocalComputationManager;
+import com.powsybl.contingency.ContingenciesProvider;
+import com.powsybl.contingency.Contingency;
+import com.powsybl.dynamicsimulation.DynamicModelsSupplier;
+import com.powsybl.dynamicsimulation.EventModelsSupplier;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.security.LimitViolationDetector;
+import com.powsybl.security.LimitViolationFilter;
+import com.powsybl.security.SecurityAnalysisReport;
+import com.powsybl.security.detectors.DefaultLimitViolationDetector;
+import com.powsybl.security.interceptors.SecurityAnalysisInterceptor;
+import com.powsybl.security.monitor.StateMonitor;
+import com.powsybl.security.strategy.OperatorStrategy;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Dynamic Security analysis main API.
+ *
+ * @author Laurent Issertial {@literal <laurent.issertial at rte-france.com>}
+ */
+public final class DynamicSecurityAnalysis {
+
+    private DynamicSecurityAnalysis() {
+        throw new IllegalStateException("Utility class should not been instantiated");
+    }
+
+    /**
+     * A dynamic security analysis runner is responsible for providing convenient methods on top of {@link DynamicSecurityAnalysisProvider}:
+     * several variants of synchronous and asynchronous run with default parameters.
+     */
+    public static final class Runner implements Versionable {
+
+        private final DynamicSecurityAnalysisProvider provider;
+
+        public Runner(DynamicSecurityAnalysisProvider provider) {
+            this.provider = Objects.requireNonNull(provider);
+        }
+
+        public CompletableFuture<SecurityAnalysisReport> runAsync(Network network,
```

Comment:
List<LimitReduction> is missing.The `LimitViolationFilter` must be removed too because it has been decided that it is the role of the application (with users expections) and not the engine (https://github.com/powsybl/powsybl-core/issues/2373). The `LimitViolationDetector` should be removed too because it is not a generic use: some simulators can do it without any provided detector.

Reply:
In general: all these `runAsync` and `run` are providing noise. The only reasons for me to provide 2 different constructors is that some engine do not support operator strategies at all. We cannot imagine a commun interface with static security analysis and some kind of a builder. It is what we do when we have not mandatory objects and a longue list.